omniport/core/base_auth/managers/get_user.py

```diff
@@ -56,20 +56,20 @@ def get_user(username):
         q_institute_webmail_address = Q(institute_webmail_address=username)
         q = (
                 q_primary_phone_number
-                | q_secondary_phone_number
                 | q_email_address
                 | q_institute_webmail_address
         )
-        contact_information = ContactInformation.objects.get(q)
+        entity_supported = ContentType.objects.get_for_model(Person)
+        try:
```

Comment:
It is a really hacky way to do this. I believe if you change the order of fields in the q itself, we are good to go. For multiple objects you may try using `.first` or if multiple objects are returned, don't return any user. We should not have something probabilistic in the code.

Reply:
order of fields in q doesnt matter as it is connected by or and will return all the objects satisfying any of the query. As for multiple multiple objects that case will only arrive if institute mail is same as webmail and in that case I have returned one having the Institute mail so i guess there is nothing that is probablistic here.